### DISCUSSION TITLE
GUI Support - WSLg Availability

### DISCUSSION THREAD

**Discussion by mattmazer0397** [WhitewaterFoundry/Pengwin#688] [Ideas]
From what I've seen you guys at Whitewater seem to do a great job of supporting GUI libraries and such in WSL (I made a related post about this previously here https://github.com/WhitewaterFoundry/Pengwin/issues/639).

I was wondering if, moving forward, anything will new will come out, or change for Pengwin?

Two points I thought up (included in the link above) are desktop wallpapers. A different user mentioned them and I was wondering if something that will roll out or be included in the future? The other point was also previously brought up in the above link, and that is desktop integration. Is xfce still the best one? Are other desktops currently supported, or will be in the future? Even yet, will pengwin-setup eventually include a full-blown GUI option where we can install libraries/desktops, etc?

Didn't think to check the blog until after I posted this (my apologies) but here's a link to a blog post about some stuff. https://www.whitewaterfoundry.com/blog

**Comment by crramirez** [comment]
Hello @mattmazer0397,

The most recent announcement was that Pengwin has built-in support for WSLg GPU acceleration from the day 1 that WSLg was released in preview builds of Windows: [GUI app support has arrived for the Windows Subsystem for Linux on the latest Windows Insiders Preview build 21362+ and Pengwin supports it out of the box](https://www.whitewaterfoundry.com/blog/2021/4/21/gui-app-support-has-arrived-for-the-windows-subsystem-for-linux-on-the-latest-windows-insiders-preview-build-21362-and-pengwin-supports-it-out-of-the-box). I recommend that you sign on our newsletters in https://www.whitewaterfoundry.com, also follow us in twitter https://twitter.com/PengwinLinux

What is coming is around this:

- Full desktop installation (Xfce, Plasma, MATE)
- Support for SystemD and snaps.
- Keep improving integration with WSLg and Windows Terminal.

You can see a presentation about it here: https://youtu.be/hmMRFV1Koyo?t=910

About which desktop, Xfce is the easiest and more stable, then we will implement Plasma and MATE. gnome-desktop is especially difficult and although some people have been able to do it, package it into an installer put a lot of challenges.

You can see the progress about desktop integration here: https://github.com/WhitewaterFoundry/pengwin-setup/pull/315. This approach uses Xserver, but we rolled back and now we are working on RDP instead. 

I finally understood what was requested about desktop wallpapers, like for example when you install Xfce in fedora and the desktop wallpaper is for Fedora. I thought that it was about Windows Terminal. We don't have plans soon about it, just because there are a lot of things in front of it.

I will check again your post and answer if there is another thing to add.

This month you will see something around Windows Terminal, stay tuned.

Regards,
Carlos
